modules/charon-core/src/main/java/org/wso2/charon3/core/schema/ServerSideValidator.java

```diff
@@ -248,11 +252,56 @@ public static AbstractSCIMObject validateUpdatedSCIMObject(AbstractSCIMObject ol
             // Check for required attributes.
             validateSCIMObjectForRequiredAttributes(newObject, resourceSchema);
         }
+
+        Map<String, Attribute> attributes = validatedObject.getAttributeList();
+
+        for (Map.Entry<String, Attribute> entry : attributes.entrySet()) {
+            String key = entry.getKey();
+            Attribute value = entry.getValue();
+
+            if (value instanceof SimpleAttribute && StringUtils.equals(key, SCIMConstants.UserSchemaConstants.LOCALE)) {
+                String localeAttributeValue = ((SimpleAttribute) value).getValue().toString();
```

Comment:
We may able to improve the logic as follows.
```
attributes.entrySet().stream()
    .filter(entry -> StringUtils.equals(entry.getKey(), SCIMConstants.UserSchemaConstants.LOCALE))
    .map(Map.Entry::getValue)
    .filter(value -> value instanceof SimpleAttribute)
    .map(value -> ((SimpleAttribute) value).getValue().toString())
    .forEach(localeAttributeValue -> {
        // Further processing of localeAttributeValue...
    });
```

Reply:
Throwing inside forEach is prohibited by IDE.

<img width="939" alt="Screenshot 2024-11-17 at 16 44 29" src="https://github.com/user-attachments/assets/02fa080e-ea79-4afc-8411-8c952602b0c7">